src/catalog/src/durable/persist.rs

```diff
@@ -1028,56 +986,140 @@ impl PersistCatalogState {
                         )));
                     }
                 }
-                StateUpdateKind::IdAllocator(key, value) => {
-                    apply(&mut self.snapshot.id_allocator, key, value, diff);
-                }
-                StateUpdateKind::IntrospectionSourceIndex(key, value) => {
-                    apply(&mut self.snapshot.introspection_sources, key, value, diff);
-                }
-                StateUpdateKind::Item(key, value) => {
-                    apply(&mut self.snapshot.items, key, value, diff);
-                }
-                StateUpdateKind::Role(key, value) => {
-                    apply(&mut self.snapshot.roles, key, value, diff);
-                }
-                StateUpdateKind::Schema(key, value) => {
-                    apply(&mut self.snapshot.schemas, key, value, diff);
-                }
-                StateUpdateKind::Setting(key, value) => {
-                    apply(&mut self.snapshot.settings, key, value, diff);
-                }
                 StateUpdateKind::StorageUsage(key, ()) => {
                     self.storage_usage_events.push((key, diff));
                 }
-                StateUpdateKind::SystemConfiguration(key, value) => {
-                    apply(&mut self.snapshot.system_configurations, key, value, diff);
-                }
-                StateUpdateKind::SystemObjectMapping(key, value) => {
-                    apply(&mut self.snapshot.system_object_mappings, key, value, diff);
-                }
-                StateUpdateKind::SystemPrivilege(key, value) => {
-                    apply(&mut self.snapshot.system_privileges, key, value, diff);
-                }
-                StateUpdateKind::Timestamp(key, value) => {
-                    apply(&mut self.snapshot.timestamps, key, value, diff);
-                }
+                _ => self.snapshot.push((kind, ts, diff)),
             }
         }
-
+        self.consolidate();
         Ok(())
     }
 
-    /// Execute and return the results of `f` on the current catalog snapshot.
+    fn consolidate(&mut self) {
+        let new_ts = self
+            .snapshot
+            .last()
```

Comment:
Is `snapshot` guaranteed to be sorted by timestamp?

Reply:
Yes. I'll add a comment and an assert here.

Reply:
Done